Backend/src/main/java/com/oops/backend/Backend/services/BackendService.java

```diff
@@ -0,0 +1,199 @@
+package com.oops.backend.Backend.services;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.sql.Timestamp;
+import java.io.File;
+import java.io.FileOutputStream;
+
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.multipart.MultipartFile;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.oops.backend.Backend.models.BucketResponse;
+import com.oops.backend.Backend.models.Solution;
+import com.oops.backend.Backend.models.Solver;
+import com.oops.backend.Backend.models.SolverBody;
+import com.oops.backend.Backend.models.TaskQueue;
+import com.oops.backend.Backend.requests.CancelSolverRequest;
+import com.oops.backend.Backend.requests.CancelTaskRequest;
+import com.oops.backend.Backend.requests.CancelUserTasksRequest;
+import com.oops.backend.Backend.requests.SolveRequest;
+
+/*
+ * Environment variables needed
+ * 
+ * SOLVER_MANAGER_ADDRESS   - Local network address of Solver manager
+ * DB_SERVICE_ADDRESS       - Address of db service 
+ * BUCKER_HANDLER_ADDRESS   - Address of bucket handler
+ */
+
+@Service
+public class BackendService {
+    private String solverManagerAddress = System.getenv("SOLVER_MANAGER_ADDRESS");
```

Comment:
Each address will consist of two different environment variables, the service name and the port. These are fetched from the "green-peace.yaml" file within the "Deployments" folder when defining a new deployment.
```suggestion
    private String solverManagerAddress = "http://" + System.getenv("SOLVER_MANAGER_SERVICE") + ":" + System.getenv("SOLVER_MANAGER_PORT");
```